### ISSUE TITLE
Providing completion in a php file that returns a array

### ISSUE THREAD

**Issue by RobinRadic** [klesun/deep-assoc-completion#129]
I would like to provide completion in a `.php` file (for example like the one below (settings.php))
I do not want to add the definition inside the file itself. I simply want to add a reference towards it.
Here's what i mean:

**settings.php**
```php
/** @return array = \App\Example::example() */
return [
    '<caret>' // CODE COMPLETION HERE
];
```

**App/Example.php**
```php
<?php
namespace App;
class Example {
    public static function example(){
        return  [
            'env'         => 'APP_NAME',
            'bind'        => 'app.name',
            'type'        => 'anomaly.field_type.text',
            'placeholder' => config('streams::distribution.name'),
            'config'      => [
                'default_value' => config('streams::distribution.name'),
            ],
        ];
    }
}
```

Is something like this possible?

**Comment by klesun** [comment]
Hi.
Maybe you would consider defining the return value in a variable? Then you will be able to reference it in phpdoc:
![image](https://user-images.githubusercontent.com/5202330/69051150-2d0e0400-0a0c-11ea-8fe8-af650403b656.png)

**Comment by RobinRadic** [comment]
I want to use this on any of the config files in laravel/pyrocms
The above notation is not used in any of the config files and require my package (a code completion provider) to modify all those files in such a way. The package is open-source/public so i'd like it to change as little as possible. It would break consistency as well.

**Comment by klesun** [comment]
Hm, ok, seems reasonable enough.
Released in [2019.11.18.002](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/update/72552)
![image](https://user-images.githubusercontent.com/5202330/69069815-cd295480-0a2f-11ea-952c-e26a88819d14.png)

**Comment by RobinRadic** [comment]
That's just awesome. Thanks a lot!
